test/tests/operations/PGP.js

```diff
@@ -0,0 +1,457 @@
+/**
+ * PGP tests.
+ *
+ * @author tlwr [toby@toby.codes]
+ *
+ * @copyright Crown Copyright 2017
+ * @license Apache-2.0
+ */
+import TestRegister from "../../TestRegister.js";
+
+var CYBERCHEF_GENERATED_KEY_PAIRS = [
+    {
+        name: "CyberChef 1",
```

Comment:
The test runner (as-is) is very hacky. It was originally built when we could only run operations in the browser. There was an entire process involving `node` calling `phantomJS` running the actual CyberChef operations, relying upon stdout and stderr piped from `phantomJS` to `node` and the test runner reporting results correctly. As expected this was a bodge and very fragile at best. (It has moved on a bit since then but is still quite bad)

It will be a good day when we remove all of our home-grown test running infrastructure and use proper test runner. This is now feasible because of @n1474335's work with node. Now we can do something like:

```
import * as CyberChef from "CyberChef";

CyberChef.bake("input", recipeConfig)
.then(x => console.log(x));
```

Instead of the `TestRegister` that is inflicted upon us currently.

The future of this file (once the test runner is refactored) is not storing key pairs in a JSON file. Generating key pairs is an operation that needs to be tested anyway. We should just generate key pairs and use those for other tests.

Again (as above) this is another area where your JS knowledge could prove useful: picking and implementing a test runner from one of the many available in the ecosystem.